docs/adr/0002-schema-first-api-design.md

```diff
@@ -0,0 +1,59 @@
+# ADR 0002: Schema-First API Design
+
+## Status
+Voorstel
+
+## Context
+Om op een consistente en duidelijke manier een goed onderhoudbare contract voor de BRP API te kunnen ontwerpen en te kunnen documenteren, is er behoefte aan een werkwijze die dit faciliteert.
+
+## Beslissing
+We hebben besloten om het contract voor de BRP API te documenteren met behulp van de [OpenAPI Specificatie][OAS] en de [Schema-First API Design][SchemaFirst] aanpak te hanteren voor de BRP API ontwikkeling.
+Dit betekent dat voor elk te implementeren wijziging/toevoeging eerst wordt gekeken of de implementatie wijziging/toevoeging zal leiden tot een wijziging van het API contract. De wijziging aan het API contract wordt dan vóór de implementatie wijziging/toevoeging uitgevoerd.
+
+## Consequenties
+- **Voordelen:**
+    - Duidelijke en consistente API contract.
+    - Verbeterde communicatie tussen de stakeholders.
+    - Eerder en gemakkelijker valideren van het API contract.
+    - Betere Developer Experience (betere documentatie, client code generatie en consumers en provider van het API contract kunnen parallel worden geïmplementeerd).
+
+- **Nadelen:**
+    - Mogelijke leercurve voor stakeholders die niet bekend zijn met Schema-First API design.
```

Comment:
Een mogelijke nadeel is de time-to-market. Code-First heeft een kortere implementatie tijd, en als het contract niet zo nauw wordt genomen (bijvoorbeeld bij interne contracten), leidt dat tot een kortere initiële time-to-market. Nadeel hiervan is wel dat andere stakeholders eventueel in de code moeten kijken om het te begrijpen.

Conclusie: Code-First kan nog best geschikt zijn voor bijvoorbeeld prototyping.

Reply:
Ik begrijp niet goed hoe de definitie van het product een leercurve voor stakeholders kan betekenen. Zijn dat externe stakeholders? Kun je daarvan een voorbeeld geven?